### PR TITLE
Update plugin for feature parity with other providers

### LINKED ISSUE #8: Fix networkd configuration
https://github.com/kontena/kontena/issues/1264

### PR DESCRIPTION
**Pull request by jakolehm**
- upgrade azure gem (fixes json error)
- fix master & agent cloud-config
- generate self-signed certificate 
- prompt for location & vm size
- open weave & openvpn ports

Fixes #8 #9

### CODE REVIEW

File: lib/kontena/machine/azure/cloudinit.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ write_files:
     content: |
       KONTENA_URI="<%= master_uri %>"
       KONTENA_TOKEN="<%= grid_token %>"
-      KONTENA_PEER_INTERFACE=eth1
+      KONTENA_PEER_INTERFACE=eth0
```
Comment (SpComb) [inline lib/kontena/machine/azure/cloudinit.yml:9]: Interesting, what does the `networkctl` look like on these machines?

File: lib/kontena/machine/azure/cloudinit.yml [hunk 2]
```diff
@@ -37,14 +37,21 @@ coreos:
...
       runtime: false
       content: |
+        # default should only match real network interfaces
         [Match]
-        Type=bridge
-        Name=weave*
-
+        Name=eth*
         [Network]
+        DHCP=yes
```
Comment (SpComb) [inline lib/kontena/machine/azure/cloudinit.yml:51]: I assume there's some coreos oem-cloudconfig that configures the `eth` interfaces, and not DHCP?

File: lib/kontena/machine/azure/cloudinit.yml (hunk 1 above)
Comment (jakolehm) [reply lib/kontena/machine/azure/cloudinit.yml:9]:
```
$ networkctl 
IDX LINK             TYPE               OPERATIONAL SETUP     
  1 lo               loopback           carrier     unmanaged 
  2 eth0             ether              routable    configured
  3 docker0          ether              no-carrier  unmanaged 
  4 datapath         ether              degraded    unmanaged 
  6 weave            ether              routable    unmanaged 
  7 dummy0           ether              off         unmanaged 
  9 vethwe-datapath  ether              degraded    unmanaged 
 10 vethwe-bridge    ether              degraded    unmanaged 
 11 vxlan-6784       ether              degraded    unmanaged 
```

File: lib/kontena/machine/azure/cloudinit.yml (hunk 2 above)
Comment (jakolehm) [reply lib/kontena/machine/azure/cloudinit.yml:51]: How to check?

**Pull request merged**